Packs/FireEyeHX/Integrations/FireEyeHXEventCollector/FireEyeHXEventCollector.py

```diff
@@ -0,0 +1,224 @@
+) -> List[Dict]:
+    """
+    Fetches events from fireye.
+    """
+
+    # incase we want to retrieve all the resolution type, no need for this parameter.
+    if resolution and resolution.lower() == 'all':
+        resolution = None
+
+    to_date = (datetime.now() + timedelta(days=1)).strftime(DATE_FORMAT)
+    filter_query = {'operator': 'between', 'arg': [first_fetch, to_date], 'field': 'reported_at'}
+
+    response = client.get_events_request(max_fetch, filter_query=json.dumps(filter_query), resolution=resolution,
+                                         min_id=min_id)
+
+    fetched_events = response.get('data', {}).get('entries', [])
+    demisto.info(f'fetched events length: ({len(fetched_events)})')
+    demisto.info(f'fetched the following events: {fetched_events}')
+
+    return fetched_events
+
+
+def main() -> None:  # pragma: no cover
+    params = demisto.params()
+    args = demisto.args()
+
+    username = params.get('credentials', {}).get('identifier', '')
+    password = params.get('credentials', {}).get('password', '')
+    base_url = params.get('url', '').rstrip('/')
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    resolution = params.get('resolution')
+    max_fetch = args.get('limit') or params.get('max_fetch')
+
+    last_run = demisto.getLastRun()
+
+    if last_run.get('last_alert_time'):
+        last_fetch = last_run.get('last_alert_time')
+    else:
+        first_fetch = params.get("first_fetch") if params.get("first_fetch") else "3 days"
+        last_fetch = arg_to_datetime(first_fetch).strftime(DATE_FORMAT)  # type: ignore[union-attr]
+
+    command = demisto.command()
+    demisto.info(f'Command being called is {command}')
+    try:
+        client = Client(
+            base_url=base_url,
+            username=username,
+            password=password,
+            verify=verify_certificate,
+            proxy=proxy,
+        )
+        if command == 'test-module':
+            return_results(test_module(client))
+        elif command == 'fetch-events':
+            events = fetch_events(
+                client=client, max_fetch=max_fetch, first_fetch=last_fetch, resolution=resolution,
+                min_id=last_run.get('last_alert_id'))
+
+            if events:
+                last_alert: dict = events[-1]
+                demisto.setLastRun({'last_alert_id': str(last_alert.get('_id')),
+                                    'last_alert_time': last_alert.get('event_at')})
+            try:
+                demisto.info(f'sending the following amount of events into XSIAM: {len(events)}')
+                send_events_to_xsiam(
+                    events=events,
+                    vendor=VENDOR,
+                    product=PRODUCT
+                )
+            except Exception as e:
+                demisto.info(f'got error when trying to send events to XSIAM: [{e}]')
```

Comment:
let's put all this in `fetch_events_command` function for more clarity and smaller `main` function